$NetBSD: patch-ab,v 1.1 2005/03/11 22:20:49 rillig Exp $

This patch makes check compile with cpp-2.95.3. The bug which it fixes
is quite subtile. When fail_if(x) is expanded, the preprocessor deletes
the last token before the ##. For cpp-2.95, this token starts with the
"o" of occured. Later versions of cpp just remove the comma.

--- src/check.h.in.orig	Tue Nov  9 14:22:16 2004
+++ src/check.h.in	Fri Mar 11 22:08:46 2005
@@ -168,7 +168,7 @@ static void __testname (void)\
 /* Fail the test case if expr is true */
 #define fail_if(expr, ...)\
         _fail_unless(!(expr), __FILE__, __LINE__,\
-        "Failure '"#expr"' occured", ## __VA_ARGS__, NULL)
+        "Failure '"#expr"' occured" , ## __VA_ARGS__, NULL)
 
 /* Always fail */
 #define fail(...) _fail_unless(0, __FILE__, __LINE__, "Failed" , ## __VA_ARGS__, NULL)
